editor/index.js

```diff
@@ -82,6 +86,15 @@ export function createEditorInstance( id, post ) {
 
 	preparePostState( store, post );
 
+	const warnDirtyBeforeUnload = ( event ) => {
```

Comment:
Actually, I'm inclined to create a separate component for this, use lifecycle to add the listeners (willMount, willUnmount) and add it to `Layout`.